.github/workflows/build.yml

```diff
@@ -0,0 +1,172 @@
+name: Build
+
+on:
+  push:
+
+env:
+  CARGO_TERM_COLOR: always
+
+jobs:
+  build-web:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: Clone
+      - uses: actions/setup-node@v1
+        name: Setup Node
+        with:
+          node-version: "*"
+      - name: Install Yarn
+        run: corepack enable
+      - name: Install npm dependencies
+        run: yarn install
+      - name: Build
+        run: yarn run build
+      - uses: actions/upload-artifact@v3
+        name: Upload Artifacts
+        with:
+          name: tauri-web
+          path: dist/*
+  build-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: Clone
+      - uses: actions-rs/toolchain@v1
+        name: Setup Rust
+        with:
+          toolchain: stable
+          override: true
+      - uses: actions/setup-node@v1
+        name: Setup Node
+        with:
+          node-version: "*"
+      - name: Install Yarn
+        run: corepack enable
+      - name: Install npm dependencies
+        run: yarn install
+      - name: Install tauri dependencies
+        run: sudo apt update && sudo apt install -y build-essential libpython3-dev libdbus-1-dev libgtk-3-dev libsoup-gnome2.4-dev libwebkit2gtk-4.0-dev
+      - name: Build Tauri
+        run: yarn run build:tauri
```

Comment:
This internally calls Rust. Due to Tauri's many dependencies, we may want to use `Swatinem/rust-cache@v1`

Reply:
Yes we can add a cache, but the build process is not bottlenecked by the dependency downloads, but by building them. We could (also) cache the build libraries, but I think a clean build would be preferable, since rust and llvm versions may change, resulting in worse binaries.

Reply:
`rust-cache` caches dependency target files as well. A different version would result in a re-build anyway

Reply:
And the rust/llvm version info is added to the cache key automatically, so no issues there

Reply:
One more addition here: just like `yarn install` should be run with a fixed .lock file, the same should be true for `cargo build`. This can be done via `yarn run build:tauri -- --locked`